vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -432,19 +466,36 @@ void TouchEventHandler::DispatchTouchEvent(TouchEventType eventType, size_t poin
   } else
 #endif // USE_FABRIC
   {
-    auto touches = folly::dynamic::array();
+    winrt::Microsoft::ReactNative::JSValueArray touches;
     for (const auto &pointer : m_pointers) {
-      folly::dynamic touch = GetPointerJson(pointer, pointer.target);
-      touches.push_back(touch);
+      touches.push_back(GetPointerJson(pointer, pointer.target));
     }
 
     // Package up parameters and invoke the JS event emitter
     const char *eventName = GetTouchEventTypeName(eventType);
     if (eventName == nullptr)
       return;
-    folly::dynamic params = folly::dynamic::array(eventName, std::move(touches), std::move(changedIndices));
-
-    m_context->CallJSFunction("RCTEventEmitter", "receiveTouches", std::move(params));
+    winrt::Microsoft::ReactNative::JSValueArray params {
+      eventName, std::move(touches), std::move(changedIndices)};
+
+    if (eventType == TouchEventType::Move || eventType == TouchEventType::PointerMove) {
+      std::wostringstream coalescingKeyStream;
+      coalescingKeyStream << eventName << m_pointers[pointerIndex].pointerId;
```

Comment:
In both of the examples here it looks like the coalescing key is derived from an integral value. I wonder if this is something we an generalize. Could save some work/allocations if we avoid string operations.